Src/Common/AppInsightsActivity.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace Microsoft.ApplicationInsights.Common
+{
+    using System;
+    using System.Diagnostics;
+    using System.Threading;
+
+    //this is a temporary solution that mimics System.DiagnosticSource,Activity and Correlation HTTP protocol:
+    //https://github.com/lmolkova/correlation/blob/master/http_protocol_proposal_v1.md
+    //It does not implement 
+    // - Request-Id length limitation and overflow
+    internal class AppInsightsActivity
+    {
+        internal static string GetRootId(string requestId)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(requestId));
+            if (requestId[0] == '|')
+            {
+                var rootEnd = requestId.IndexOf('.');
+                var rootId = requestId.Substring(1, rootEnd - 1);
+                return rootId;
+            }
+            return requestId;
+        }
+
+
+        internal static string GenerateNewId()
+        {
+            if (_machinePrefix == null)
+                Interlocked.CompareExchange(ref _machinePrefix, Environment.MachineName + "-" + ((int)Stopwatch.GetTimestamp()).ToString("x"), null);
```

Comment:
well the point of machine name is to give a bit more info to human being that is looking into telemetry Id. 
In Azure, machine name looks like `RD0003FF21406D` that is not long.

However I agree that machine name in a huge cluster may not be that useful.